src/components/views/MovieCard.vue

```diff
@@ -1,99 +1,46 @@
 <script>
 import { defineComponent } from 'vue';
-import AppTag from '@/components/common/AppTag.vue';
-import SectionTitle from '@/components/common/SectionTitle.vue';
+import AppTag from '@/components/common/App/AppTag.vue';
+import MovieLength from '@/components/common/Movie/MovieLength.vue';
+import MovieTitle from '@/components/common/Movie/MovieTitle.vue';
+import MoviePoster from '@/components/common/Movie/MoviePoster.vue';
 
 export default defineComponent({
-    components: { AppTag, SectionTitle },
-    props: {
-        movie: {
-          type:Object,
-          required: true,
-        }
-    },
-    data() {
-        return {
-            movieData: this.movie,
-        };
-    },
-    methods: {
-        durationInHours(durationInMinutes) {
-            const hours = Math.floor(durationInMinutes / 60);
-            const minutes = `0${durationInMinutes % 60}`.slice(-2);
-            return `${hours} h ${minutes} min`;
-        }
-    },
+  components: { AppTag, MovieLength, MovieTitle, MoviePoster },
+  props: {
+    movie: {
+      type:Object,
+      required: true,
+    }
+  },
 });
 </script>
 
 <template>
-    <div :class="[`movie-card card-${movieData.id}`]" >
-        <p class="movie-title" >{{ movieData.title }}</p>
-        <p class="movie-duration">{{durationInHours(movieData.length)}}</p>
-        <div class="movie-poster">
-            <img :src="movieData.poster_url" alt="movie poster" />
-        </div>
-        <AppTag>{{movieData.genre.name}}</AppTag>
+  <div class="movie-card" >
+    <MovieTitle>{{ movie.title }}</MovieTitle>
+    <div>
+      <MovieLength :lengthInMinutes="movie.length" />
+      <MoviePoster :src="movie.poster_url" />
+      <AppTag>{{movie.genre.name}}</AppTag>
     </div>
```

Comment:
(for the future)

Good to see that you are using components with ease and concepts like slots. Well done here! 🔥
In this case moving all code outside doesn't feel necessary. With time going you will get feeling when it is too busy and when to encapsulate code 🙂

Reply:
You mean it wasn't worth to extract MoviePoster, MovieLength and MovieTitle components? I did it only when it appeared that I will need to re-use their styles several times in the project (see screenings cards and movie cards here: https://www.figma.com/file/m65uaTJoACt8ZTKGPbKHol/VUE-bootcamp-%E2%80%94-montecinema?node-id=0%3A1)

Reply:
If so, then it is good move, when you will use them across the app 👍